driver-api/src/main/java/io/openmessaging/benchmark/driver/BenchmarkConsumer.java

```diff
@@ -18,6 +18,15 @@
  */
 package io.openmessaging.benchmark.driver;
 
-public interface BenchmarkConsumer extends AutoCloseable {
+import java.util.concurrent.CompletableFuture;
 
+public interface BenchmarkConsumer extends AutoCloseable {
+    /**
+     * Start listening for messages and return a callback to track the completion of the
+     * operation.
+     *
+     * @param callback
+     * @return
+     */
+    CompletableFuture<Void> receiveAsync(ConsumerCallback callback);
```

Comment:
Is this method intended to defer the creation of the consumer and starting the message reception?

Reply:
I was pulling the callback out of the constructor and into the interface. They way we don't break pulsar consumer when we implement it.